scripts/factory-agent.sh

```
@@ -1119,6 +1235,385 @@ pr_failing_advisory() {
+    jq -nc --argjson skipped "$skipped_json" --argjson manual "$manual" \
+      '{ dispatched: [], skipped: $skipped, manualCommands: $manual }'
+    return 0
+  fi
+  # dispatch-premerge resolves each lane's root: mobile from --repo-root, desktop
+  # from --desktop-root (fossil-asserted). Only pass lanes whose root is ready.
+  [[ -z "${mobile_root:-}" ]] && lanes=$(echo "$lanes" | sed -E 's/(^|,)mobile(,|$)/\1\2/; s/^,|,$//')
+  [[ -z "${desktop_root:-}" ]] && lanes=$(echo "$lanes" | sed -E 's/(^|,)desktop(,|$)/\1\2/; s/^,|,$//')
+  root_flags="--repo-root ${mobile_root:-$REPO_ROOT}"
+  [[ -n "${desktop_root:-}" ]] && root_flags="$root_flags --desktop-root $desktop_root"
+
+  # shellcheck disable=SC2086 -- root_flags is a deliberately word-split flag list
```

Shellcheck (0.11.0)

[error] Couldn't parse this shellcheck directive. Fix to allow more checks.

(SC1073)